### PR TITLE
DT-623: create initial password link expired - new link automatically…

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**
… issued

### CODE REVIEW

File: src/main/resources/templates/createPasswordExpired.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <h1 class="govuk-panel__title">Your create a password link has expired</h1>
+  </div>
+  <div>
+
+    <p class="govuk-body">The link you have selected has now expired. For security, the link is only valid for 7
+      days.</p>
+    <p class="govuk-body">We have sent you a new link, You need to use this link within 7 days to create a password.</p>
+    <p th:if="${link}" class="govuk-body">
+      <br>
+      <code>application.smoketest.enabled</code> environment variable is set.
```
Comment (andymarke) [inline src/main/resources/templates/createPasswordExpired.html:22]: Curious, how does this *not* get displayed in prod? "environment variable is set"

Comment (SimonMitchellMOJ) [reply src/main/resources/templates/createPasswordExpired.html:22]: The link should only be added to the view/page if smoketest variable is set to true, I will check the method is doing this correctly

File: src/main/resources/static/css/app-ie8.css [hunk 2]
```diff
@@ -67,9 +67,13 @@
 }
 
 .auth-card__content p {
-  margin-bottom: 0;
+    margin-bottom: 0;
```
Comment (petergphillips) [inline src/main/resources/static/css/app-ie8.css:70]: rest of class is 2 spaces, not 4

File: src/main/resources/static/css/app.css [hunk 3]
```diff
@@ -1,91 +1,95 @@
 #header-container {
-  border-bottom-color: white;
+    border-bottom-color: white;
```
Comment (petergphillips) [inline src/main/resources/static/css/app.css:2]: should be 2 spaces for css

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/ResetPasswordController.kt [hunk 4]
```diff
@@ -32,7 +32,7 @@ class ResetPasswordController(private val resetPasswordService: ResetPasswordSer
                               private val telemetryClient: TelemetryClient,
                               @param:Value("\${application.smoketest.enabled}") private val smokeTestEnabled: Boolean,
                               @Value("\${application.authentication.blacklist}") passwordBlacklist: Set<String?>?) :
-    AbstractPasswordController(resetPasswordService, tokenService, userService, telemetryClient, "resetPassword", "setPassword", passwordBlacklist) {
+    AbstractPasswordController(resetPasswordService, tokenService, userService, telemetryClient, "resetPassword", "initialPassword", passwordBlacklist) {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/ResetPasswordController.kt:35]: these changes don't look right - reset password failure shouldn't take the user to the initial password page

File: src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/ResetPasswordSpecification.groovy [hunk 5]
```diff
@@ -8,7 +8,13 @@ import static uk.gov.justice.digital.hmpps.oauth2server.integration.specs.model.
 class ResetPasswordSpecification extends DeliusIntegrationSpec {
 
     def "A user can cancel reset password"() {
-        given: 'I would like to reset my password'
+        given: 'log in and out a user to ensure that no user is logged in for test'
+        to LoginPage
+        loginAs ITAG_USER, 'password'
+        at HomePage
+        logout()
```
Comment (petergphillips) [inline src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/ResetPasswordSpecification.groovy:15]: should really just have a `before` that does a `logout`

File: src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/pages/SetPasswordPage.groovy [hunk 6]
```diff
@@ -4,7 +4,7 @@ import geb.Page
 
 class SetPasswordPage extends Page {
 
-    static url = '/auth/set-password'
+    static url = '/auth/initial-password'
```
Comment (petergphillips) [inline src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/pages/SetPasswordPage.groovy:7]: this is incorrect - should be `set-password` as different flows / pages for create password and reset password

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/AbstractPasswordControllerTest.kt [hunk 7]
```diff
@@ -70,7 +76,7 @@ class AbstractPasswordControllerTest {
     setupCheckAndGetTokenValid()
     setupGetUserCallForProfile(null)
     val modelAndView = controller.setPassword("d", "@fewfewfew1", "@fewfewfew1", null)
-    assertThat(modelAndView.viewName).isEqualTo("setPassword")
+    assertThat(modelAndView.viewName).isEqualTo("initialPassword")
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/AbstractPasswordControllerTest.kt:79]: should be `setPassword` here as based off the reset password controller

**Pull request merged**